Assets/Scripts/PlayerCharacter/PlayerCharacterController.cs

```diff
@@ -44,7 +55,15 @@ private IEnumerator DisableInputCoroutine(float seconds)
             disabled = true;
             yield return new WaitForSeconds(seconds);
             disabled = false;
-            currentCoroutine = null;
+            //currentCoroutine = null;
+            currentCoroutine = StartCoroutine(HealthDecreaseCoroutine(healthDecreasingTiming));
+        }
+
+        private IEnumerator HealthDecreaseCoroutine(float seconds)
+        {
+            combatEntity.GetDecreased(heathDecreasingValue);
+            yield return new WaitForSeconds(seconds);
+            currentCoroutine = StartCoroutine(HealthDecreaseCoroutine(healthDecreasingTiming));
```

Comment:
Starting a new coroutine is a relatively expensive operation compared to simple looping and yielding from coroutines. It is advised to use a while loop with condition `true` to create an infinite loop.

For example,
```csharp
private IEnumerator SomeCoroutine(float interval)
{
    while (true)
    {
        yield return new WaitForSeconds(interval);
        DoSomething();
    }
}
```

Or for better performance, you can also cache the `WaitForSeconds` object.
```csharp
private IEnumerator SomeCoroutine(float interval)
{
    WaitForSeconds intervalYielder = new WaitForSeconds(interval);

    while (true)
    {
        yield return intervalYielder;
        DoSomething();
    }
}
```